core/eth/tx.go

```diff
@@ -280,26 +282,38 @@ func (t *Transactor) RegisterOperatorWithChurn(
 	return nil
 }
 
-// DeregisterOperator deregisters an operator with the given public key from the all the quorums that it is
+// DeregisterOperator deregisters an operator with the given public key from the specified the quorums that it is
 // registered with at the supplied block number. To fully deregister an operator, this function should be called
 // with the current block number.
-func (t *Transactor) DeregisterOperator(ctx context.Context, pubkeyG1 *core.G1Point, blockNumber uint32) error {
+// If the operator isn't registered with any of the specified quorums, this function will return error, and
+// no quorum will be deregistered.
+func (t *Transactor) DeregisterOperator(ctx context.Context, pubkeyG1 *core.G1Point, blockNumber uint32, quorumIds []core.QuorumID) error {
+	if len(quorumIds) == 0 {
```

Comment:
Can we dedup them as well? Just wanted to make sure we don't loop over a massive slice on :305

Reply:
That doesn't seem to help much, no matter it's dup or no dup, it's provided by users who are going to run this cmd

Reply:
Sure, how about just check `len(quorumIDs) < 256`? It would prevent them from passing a huge number of elems

Reply:
still to be done

Reply:
https://github.com/Layr-Labs/eigenda/pull/381/commits/1629ea6e15a843a9bd3eb58e38bb7652a6da81e6